lib/licensed/sources/gradle.rb

```diff
@@ -122,61 +76,121 @@ def configurations
         end
       end
 
-      def self.add_gradle_license_report_plugins_block(gradle_build_file)
+      # Prefixes the gradle command with the project name for multi-build projects.
+      def format_command(command)
+        if config.source_path != config.pwd
+          project = File.basename(config.source_path)
+        end
+        project.nil? ? command : "#{project}:#{command}"
+      end
 
-        if gradle_build_file.include? "plugins"
-          gradle_build_file.gsub(/(?<=plugins)\s+{/, " { id 'com.github.jk1.dependency-license-report' version '1.16'")
-        else
+      # Returns a key to uniquely identify a name and version in the obtained CSV content
+      def csv_key(name:, version:)
+        "#{name}-#{version}"
+      end
 
-          gradle_build_file = " plugins { id 'com.github.jk1.dependency-license-report' version '1.16' }" + gradle_build_file
+      # Loads a license report CSV data as a hash of :name-:version => :url pairs
+      # Returns a hash of dependency identifiers to their license content URL
+      def load_csv
+        begin
+          gradle_licenses_dir = File.join(config.root, GRADLE_LICENSES_PATH)
+          CSV.foreach(File.join(gradle_licenses_dir, GRADLE_LICENSES_CSV_NAME), headers: true).each_with_object({}) do |row, hsh|
+            name, _, version = row["artifact"].rpartition(":")
+            key = csv_key(name: name, version: version)
+            hsh[key] = row["moduleLicenseUrl"]
+          end
+        ensure
+          FileUtils.rm_rf(gradle_licenses_dir)
         end
       end
 
-      def self.gradle_command(*args, path:, executable:, configurations:)
-        gradle_build_file = File.read("build.gradle")
+      # Returns the cached url for the given dependency
+      def url_for(dependency)
+        @csv[csv_key(name: dependency.name, version: dependency.version)]
+      end
 
-        if !gradle_build_file.include? "dependency-license-report"
-          gradle_build_file = Licensed::Sources::Gradle.add_gradle_license_report_plugins_block(gradle_build_file)
+      # The Gradle::Runner class is a wrapper which provides
+      # an interface to run gradle commands with the init script initialized
+      class Runner
+        def initialize(root_path, configurations)
+          @root_path = root_path
+          @configurations = configurations
+          @executable = executable
         end
 
-        Dir.chdir(path) do
-          Tempfile.create(["license-", ".gradle"], path) do |f|
-            f.write(gradle_build_file)
-            f.write gradle_file(configurations)
-            f.close
-            Licensed::Shell.execute(executable, "-q", "-b", f.path, *args)
+        def run(*args)
+          Dir.chdir(@root_path) do
+            Tempfile.create(["init", ".gradle"], @root_path) do |f|
+              f.write(init_script(@configurations))
+              f.close
+              args << "--no-configuration-cache" if gradle_version >= "6.6"
```

Comment:
The [configuration cache](https://docs.gradle.org/current/userguide/configuration_cache.html) is an incubating feature that can be activated manually. The gradle plugin for licenses does not support it so we prevent it to run for gradle version supporting it.